spec.md

```diff
@@ -0,0 +1,948 @@
+
+```
+Required:     No
+Multi-Valued: No
+Syntax:       [ minus ] 1*DIGIT
+Since:        0.2.0
+Deprecated:   0.3.0
+Removed:      1.0.0
+```
+
+The `NOTESGAP` header is deprecated and MUST NOT be used.
+Implementations MUST ignore the field if present.
+
+> [!NOTE]
+>
+> The `NOTESGAP` header was defined in version 0.1.0 But its semantics were never fully specified.
+
+### 3.12. The `START` and `END` Headers
+
+```
+Required:     No
+Multi-Valued: No
+Syntax:       1*DIGIT
+Since:        0.2.0
+```
+
+The `START` and `END` header specify two time points in milliseconds relative to the start of the audio data that indicate a start and end point for the song.
+Game implementations SHOULD start and end the song at the specified points and scale scoring accordingly.
+Both `START` and `END` values are integers.
+
+> [!NOTE]
+>
+> The `START` and `END` values do not affect the placement of notes nor any other time codes relative to the audio.
+> They simply indicate that a song should be started or stopped a certain amount of time into the audio file.
+
+> [!WARNING]
+>
+> **Breaking Change** in version 2.0.0
+>
+> In versions 0.x and 1.x `START` was specified in seconds as a floating point number.
+> Version 2.0.0 changes this to an integer and milliseconds.
+
+### 3.13. The `PREVIEWSTART` Header
+
+```
+Required:     No
+Multi-Valued: No
+Syntax:       1*DIGIT
+Since:        0.2.0
+```
+
+The `PREVIEWSTART` header indicates a time offset in milliseconds relative to the start of the audio where the preview starts.
+Implementations MAY use this value when playing a song in a preview setting (e.g. during song selection).
+In its absence implementations SHOULD default to the start of the medley section (if available).
+
+> [!WARNING]
+>
+> **Breaking Change** in version 2.0.0
+>
+> In versions 0.x and 1.x `PREVIEWSTART` was specified in seconds as a floating point number.
+> Version 2.0.0 of this specification changes this to an integer and milliseconds.
+
+### 3.14. The `MEDLEYSTART` and `MEDLEYEND` Headers
+
+```
+Required:     No
+Multi-Valued: No
+Syntax:       1*DIGIT
+Since:        2.0.0
+```
+
+The `MEDLEYSTART` and `MEDLEYEND` headers indicate in milliseconds the start and end of the medley section of a song relative to the start of the audio.
+These tags replace `MEDLEYSTARTBEAT` and `MEDLEYENDBEAT` in version 2.0.0 of this specification.
+
+### 3.15. The `MEDLEYSTARTBEAT` and `MEDLEYENDBEAT` Headers
+
+```
+Required:     No
+Multi-Valued: No
+Syntax:       1*DIGIT
+Since:        0.2.0
+Removed:      2.0.0
+```
+
+The `MEDLEYSTARTBEAT` and `MEDLEYENDBEAT` headers indicate in beats the start and end of the medley section of a song.
+Implementations MUST respect the `GAP` value when calculating the medley start and end times.
+
+> [!WARNING]
+>
+> **Breaking Change** in version 2.0.0
+>
+> The headers `MEDLEYSTARTBEAT` and `MEDLEYENDBEAT` have been replaced by `MEDLEYSTART` and `MEDLEYEND` in version 2.0.0 of this specification.
```

Comment:
I'd be ok with that, but I think that would be sending mixed signals. We could either:
- Deprecate `MEDLEYSTARTBEAT` and `MEDLEYENDBEAT` with no replacement (and then add `MEDLEYSTART` and `MEDLEYEND` in v2)
- Deprecate `MEDLEYSTARTBEAT` and `MEDLEYENDBEAT` and introduce `MEDLEYSTART` and `MEDLEYEND` in 1.2.0.

In the first case it could look like the medley feature in general is being deprecated (which is not the intent). In the second case we would have to deal with files where both `MEDLEYSTARTBEAT` and `MEDLEYSTART` are present (like with `MP3` and `AUDIO`).

I would prefer if we would remove `MEDLEYSTARTBEAT` and `MEDLEYENDBEAT` without prior deprecation.

Reply:
I agree with codello we should remove MEDLEYSTARTBEAT and MEDLEYENDBEAT without prior deprication. cause we keep the medley feature.